api/geth_backend.go

```diff
@@ -692,6 +696,88 @@ func (b *GethStatusBackend) ConvertToKeycardAccount(account multiaccounts.Accoun
 	return nil
 }
 
+func (b *GethStatusBackend) CreateAccountAndLogin(request *requests.CreateAccount) error {
+
+	if err := request.Validate(); err != nil {
+		return err
+	}
+
+	if err := b.accountManager.InitKeystore(filepath.Join(request.BackupDisabledDataDir, "keystore")); err != nil {
+		return err
+	}
+
+	b.UpdateRootDataDir(request.BackupDisabledDataDir)
+	err := b.OpenAccounts()
+	if err != nil {
+		b.log.Error("failed open accounts", err)
+		return err
+	}
+
+	generator := b.accountManager.AccountsGenerator()
+
+	generatedAccountInfos, err := generator.Generate(12, 1, "")
+
+	if err != nil {
+		return err
+	}
+
+	info := generatedAccountInfos[0]
+
+	derivedAddresses, err := generator.DeriveAddresses(info.ID, paths)
+	if err != nil {
+		return err
+	}
+
+	_, err = generator.StoreDerivedAccounts(info.ID, "", paths)
+	if err != nil {
+		return err
+	}
+
+	account := multiaccounts.Account{
+		KeyUID:        info.KeyUID,
+		KDFIterations: sqlite.ReducedKDFIterationsNumber,
+	}
+
+	settings, err := defaultSettings(info, derivedAddresses, nil)
+	if err != nil {
+		return err
+	}
+
+	nodeConfig, err := defaultNodeConfig(settings.InstallationID, request)
+	if err != nil {
+		return err
+	}
+
+	walletDerivedAccount := derivedAddresses[pathDefaultWallet]
+	walletAccount := &accounts.Account{
+		PublicKey: types.Hex2Bytes(walletDerivedAccount.PublicKey),
+		KeyUID:    info.KeyUID,
+		Address:   types.HexToAddress(walletDerivedAccount.Address),
+		Color:     "",
+		Wallet:    true,
+		Path:      pathDefaultWallet,
+		Name:      "Ethereum account",
```

Comment:
is there no constants for strings like these? 🤔

Reply:
I can const at the top of the file